File: snippets/swift/BreezSDKExamples/Sources/GettingStarted.swift
import BreezSDKLiquid


func gettingStarted() throws -> BindingLiquidSdk? {
    // ANCHOR: init-sdk
    let mnemonic = "<mnemonic words>"

    // Create the default config, providing your Breez API key
    var config = try defaultConfig(network: LiquidNetwork.mainnet, breezApiKey: "<your-Breez-API-key>")

    // Customize the config object according to your needs
    config.workingDir = "path to an existing directory"

    let connectRequest = ConnectRequest(config: config, mnemonic: mnemonic)
    let sdk = try? connect(req: connectRequest)
    // ANCHOR_END: init-sdk

    return sdk
}


func gettingStartedNodeInfo(sdk: BindingLiquidSdk) {
    // ANCHOR: fetch-balance
    if let info = try? sdk.getInfo() {
        let balanceSat = info.walletInfo.balanceSat
        let pendingSendSat = info.walletInfo.pendingSendSat
        let pendingReceiveSat = info.walletInfo.pendingReceiveSat

        print(balanceSat)
        print(pendingSendSat)
        print(pendingReceiveSat)
    }
    // ANCHOR_END: fetch-balance
}

// ANCHOR: logging
class SDKLogger: Logger {
    func log(l: LogEntry) {
        print("Received log [", l.level, "]: ", l.line)
    }
}

func logging() throws {
    try? setLogger(logger: SDKLogger())
}
// ANCHOR_END: logging

// ANCHOR: add-event-listener
class SDKEventListener: EventListener {
    func onEvent(e: SdkEvent) {
        print("Received event: ", e)
    }
}

func addEventListener(sdk: BindingLiquidSdk, listener: SDKEventListener) throws -> String? {
    let listenerId = try? sdk.addEventListener(listener: listener)
    return listenerId
}
// ANCHOR_END: add-event-listener

// ANCHOR: remove-event-listener
func removeEventListener(sdk: BindingLiquidSdk, listenerId: String) throws {
    try? sdk.removeEventListener(id: listenerId)
}
// ANCHOR_END: remove-event-listener

// ANCHOR: disconnect
func disconnect(sdk: BindingLiquidSdk) throws {
    try? sdk.disconnect()
}
// ANCHOR_END: disconnect

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/GettingStarted.kt
package com.example.kotlinmpplib

import breez_sdk_liquid.*
class GettingStarted {
    fun start() {
        // ANCHOR: init-sdk
        val mnemonic = "<mnemonic words>"

        // Create the default config, providing your Breez API key
        val config : Config = defaultConfig(LiquidNetwork.MAINNET, "<your Breez API key>")

        // Customize the config object according to your needs
        config.workingDir = "path to an existing directory"

        try {
            val connectRequest = ConnectRequest(config, mnemonic)
            val sdk = connect(connectRequest)
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: init-sdk
    }

    fun fetchBalance(sdk: BindingLiquidSdk) {
        // ANCHOR: fetch-balance
        try {
            val info = sdk.getInfo()
            val balanceSat = info?.walletInfo.balanceSat
            val pendingSendSat = info?.walletInfo.pendingSendSat
            val pendingReceiveSat = info?.walletInfo.pendingReceiveSat
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: fetch-balance
    }

    // ANCHOR: logging
    class SDKLogger : Logger {
        override fun log(l: LogEntry) {
            // Log.v("SDKListener", "Received log [${l.level}]: ${l.line}")
        }
    }

    fun setLogger(logger: SDKLogger) {
        try {
            setLogger(logger)
        } catch (e: Exception) {
            // handle error
        }
    }
    // ANCHOR_END: logging

    // ANCHOR: add-event-listener
    class SDKListener : EventListener {
        override fun onEvent(e: SdkEvent) {
            // Log.v("SDKListener", "Received event $e")
        }
    }

    fun addEventListener(sdk: BindingLiquidSdk, listener: SDKListener): String? {
        try {
            val listenerId = sdk.addEventListener(listener)
            return listenerId
        } catch (e: Exception) {
            // handle error
            return null
        }
    }
    // ANCHOR_END: add-event-listener

    // ANCHOR: remove-event-listener
    fun removeEventListener(sdk: BindingLiquidSdk, listenerId: String)  {
        try {
            sdk.removeEventListener(listenerId)
        } catch (e: Exception) {
            // handle error
        }
    }
    // ANCHOR_END: remove-event-listener

    // ANCHOR: disconnect
    fun disconnect(sdk: BindingLiquidSdk)  {
        try {
            sdk.disconnect()
        } catch (e: Exception) {
            // handle error
        }
    }
    // ANCHOR_END: disconnect
}
